limitador-server/src/main.rs

```diff
@@ -265,12 +265,18 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
 
     let limiter = Arc::clone(&rate_limiter);
     let handle = Handle::current();
-    // it should not fail because the limits file has already been read
-    let limits_file_dir = Path::new(&limit_file).parent().unwrap();
-    let limits_file_path_cloned = limit_file.to_owned();
+
     // structure needed to keep state of the last known canonical limits file path
     let mut last_known_canonical_path = fs::canonicalize(&limit_file).unwrap();
 
+    // This isn't probably the best fix, see https://github.com/Kuadrant/limitador/issues/119
+    let mut parent = Path::new(&limit_file).parent().unwrap();
+    if !parent.is_absolute() {
+        parent = last_known_canonical_path.as_path().parent().unwrap();
```

Comment:
This is approach will not work in k8s. In k8s the file structure is as follows:

```
/home/limitador/bin $ ls -al /home/limitador/etc                                                     
total 0                                                                                              
drwxrwsrwx    3 root     10007600       100 Jul 20 22:25 .                                           
drwxr-sr-x    1 limitado limitado        17 Jul 20 22:25 ..                                          
drwxr-sr-x    2 root     10007600        47 Jul 20 22:25 ..2022_07_20_22_25_07.358335008             
lrwxrwxrwx    1 root     10007600        31 Jul 20 22:25 ..data -> ..2022_07_20_22_25_07.358335008                    
lrwxrwxrwx    1 root     10007600        28 Jul 20 22:25 limitador-config.yaml -> ..data/limitador-config.yaml
```

Thus, `last_known_canonical_path` would have a value similar (in structure) as `/home/limitador/etc/..2022_07_20_22_25_07.358335008/limitador-config.yaml`. The parent would be `/home/limitador/etc/..2022_07_20_22_25_07.358335008`, and that is not the directory we want to monitor. We want to monitor the `/home/limitador/etc` directory.

Reply:
I mean that in k8s having a relative "local to the current directory" approach would not work. But using the operator, this is never the case, as the operator always uses absolute path (possibly full of symlinks)